lecture-notes/3-bld.md

```diff
@@ -323,3 +328,156 @@ Make has been around a long time and its performance and correctness are
 well understood. Its interface is well understood, too; people who know Make
 are at home in most other projects that use Make. Not so for a custom shell
 script.
+
+### Variables
+
+Up until now we have written rules and filenames and commands as constants in
+the Makefile. This means that if you ever had to reference a file `foo.c` more
+than once in the Makefile, you would have to copy around the text. Fortunately,
+like other programming languages, Make provides variables.
+
+User-defined variables come in two forms in Make: *simply expanded* variables
+use `NAME := val` syntax and *recursively expanded* variables use `NAME = val`
+syntax. You will likely not want to use recursively expanded variables much, if
+at all, as they have different semantics than in other programming languages:
+
+**Simply expanded** means that variable references and function calls on the
```

Comment:
nit: should we hyphenate "simply-expanded"?

Reply:
no idea. if this is correct, please say so